src/bridge.js

```diff
@@ -62,3 +123,28 @@ exports.bridge = async function(port, config, registration) {
     bridge.listen(port, config);
     log.info("Matrix-side listening on port:", port);
 };
+
+/**
+ * Returns the email address obtained from Matrix UserId
+ * @param userId        userId in the format `@_email_<localPart>_<domain.tld>:localhost`.
+ * @param homeServer    homeServer address.
+ * @returns {string}    Email-ID obtained from bridge userID `localPart@domain.tld`.
+ */
+function getMailIdFromUserId(userId, homeServer) {
+    const email = userId.slice(8, userId.lastIndexOf(":"+ homeServer));
+    const localPart = email.slice(0, email.lastIndexOf('_'));
```

Comment:
https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/src/models/users/matrix.ts#L16 will parse the localpart for you, and I'd probably prefer using that since we have tests to ensure accuracy :)